hasjob/static/js/app.js

```diff
@@ -153,11 +165,99 @@ window.Hasjob.StickieList = {
+    context.rect(0, 0, canvas.width, canvas.height);
+    var grd = context.createLinearGradient(0, 0, canvas.width, canvas.height);
+
+    grd.addColorStop(1, '#DF3499');
+    grd.addColorStop(0.7, '#E05F26');
+    grd.addColorStop(0.5, '#DF5C2A');
+    grd.addColorStop(0.1, '#F1D564');
+    grd.addColorStop(0, '#FFFFA2');
+
+    context.fillStyle = grd;
+    context.fill();
+    //Store the canvas context and end colour of the conversion funnel, later to be used by window.Hasjob.Util.setFunnelColour for picking the colour for a conversion funnel value for a job post.
+    window.Hasjob.Config[funnelName] = {};
+    window.Hasjob.Config[funnelName].canvasContext = context;
+    window.Hasjob.Config[funnelName].maxColour = '#DF3499';
+  },
+  /*
+    Picks the colour for the value from the colour gradient canvas based on a scale of 0 to maxValue.
+    Takes 'funnelName', value, elementId'
+    'funnelName' - conversion funnel's name.
+    'value' - conversion funnel value for the job post
+    'elementId' - id attribute of the element of which background colour is to be set
+  */
+  setGradientColour: function(funnelName, value, elementId) {
+    //rgba - RGBA values at a particular point in the canvas.
+    var rgba = window.Hasjob.Config[funnelName].canvasContext.getImageData(value, 1, 1, 1).data;
+    if (rgba[0] > 255 || rgba[1] > 255 || rgba[2] > 255) {
+      // rgb value is invalid, hence return white
+      colourHex ="#FFFFFF";
+    } else if (rgba[0] == 0 && rgba[1] == 0 && rgba[2] == 0) {
+      // value greater than maxValue hence return the last colour of the gradient
+      colourHex = window.Hasjob.Config[funnelName].maxColour;
+    } else {
+      // Get the colour code in hex from RGB values returned by getImageData
+      colourHex = "#" + (("000000" + (rgba[0] << 16) | (rgba[1] << 8) | rgba[2]).toString(16)).slice(-6);
+    }
+    // Set the background colour of the element
+    var element = document.getElementById(elementId);
+    element.classList.add("funnel-color-set");
+    element.style.backgroundColor = colourHex;
+  },
+  renderGradientColour: function() {
+    $('.js-funnel').each(function() {
+      if(!$(this).hasClass("funnel-color-set")) {
+        Hasjob.StickieList.setGradientColour($(this).data('funnel-name'), $(this).data('funnel-value'), $(this).attr('id'));
+      }
+    });
+  },
+  createGradientColour: function() {
+    Hasjob.StickieList.createGradientColourScale('impressions', Hasjob.Config.MaxFunnelStat.max_impressions);
+    Hasjob.StickieList.createGradientColourScale('views', Hasjob.Config.MaxFunnelStat.max_views);
+    Hasjob.StickieList.createGradientColourScale('opens', Hasjob.Config.MaxFunnelStat.max_opens);
+    Hasjob.StickieList.createGradientColourScale('applied', Hasjob.Config.MaxFunnelStat.max_applied);
+  },
+  funnelStatInit: function() {
+    window.addEventListener('onStickiesInit', function (e) {
+      if (window.Hasjob.Config.MaxFunnelStat) {
```

Comment:
All the three event listeners could be under this one `if` condition, since they can't be registered independently anyway.